Source/ACE.Server/WorldObjects/Portal.cs

```diff
@@ -134,6 +134,77 @@ public override ActivationResult CheckUseRequirements(WorldObject activator)
                     // You are too powerful to interact with that portal!
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.YouAreTooPowerfulToUsePortal));
                 }
+
+                var playerPkLevel = player.PkLevel;
+
+                if (PropertyManager.GetBool("pk_server").Item)
+                    playerPkLevel = PKLevel.PK;
+                else if (PropertyManager.GetBool("pkl_server").Item)
+                    playerPkLevel = PKLevel.PKLite;
+
+                if ((PortalRestrictions & PortalBitmask.Undef) != 0)
```

Comment:
This line doesn't make much sense as written. You are doing something & 0 != 0, which is impossible with a standard bitmask... the only way this might be possible is if PortalRestrictions == null, in which case, you should just be writing PortalRestrictions == null